### PR TITLE
DHFPROD-4216: Collapse All/Expand All buttons for Source and Entity tables in Mapping UI.

### PR DESCRIPTION
**Pull request by xnikhil08**
### Description
This PR enables the Collapse All and Expand All buttons in Source and Entity tables in mapping UI.
Also, it fixes the bug where the nested levels used to collapse when navigating between source documents, using the navigation buttons.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by xnikhil08** [comment]
> @xnikhil08, I'm seeing some issues with Collapse All. See in my screencast when the Collapse All click only collapses the nested zip property and nothing else:
> 
> https://drive.google.com/file/d/1FEzflZKugfPbTSclvCBt2PleHgJM-bP9/view
> 
> I would expect all the properties to collapse?

@wooldridge - Collapse All would close everything except the root and first nested level, as per the requirement.

**Comment by bsrikan** [comment]
It could get a little confusing @wooldridge . Please checkout the test plan and let me know if it makes sense.

**Comment by wooldridge** [comment]
@bsrikan, thanks. I went over the test plan for 4216 and tried to test the list: https://wiki.marklogic.com/display/ENGINEERING/Test+Plan+for+Mapping+Epic

Here's what I see:

- Verify the default behavior when top 2 levels are expanded -> Button text should say "Expand All" **OK**
- Verify by clicking on "Expand All" button -> All the levels should be expanded with indentation **OK**
- Verify button text when all the levels are expanded -> Button text should say "Collapse All" **OK**
- Verify by clicking on "Collapse All" button -> All but top 2 levels should be collapsed **OK**
- Verify button text when all 1st level properties are collapsed -> Button text should say "Collapse All" **NO** (Button says "Expand All," as it should in my opinion) See: https://drive.google.com/file/d/1eSE_xpsw86tZpja8hDWFbXlNo7qsFghT/view
- Verify button text when > *top 2 level properties are collapsed -> Button text should say "Collapse All" **OK**
- Verify button text when > *top 2 level properties are expanded -> Button text should say "Expand All" **NO** (I'm seeing "Collapse All" at times when 3rd level are expanded) See: https://drive.google.com/file/d/1eSE_xpsw86tZpja8hDWFbXlNo7qsFghT/view

But, honestly, I'm not sure if those two above that fail make sense to me.

**Comment by xnikhil08** [comment]
> @bsrikan, thanks. I went over the test plan for 4216 and tried to test the list: https://wiki.marklogic.com/display/ENGINEERING/Test+Plan+for+Mapping+Epic
> 
> Here's what I see:
> 
> * Verify the default behavior when top 2 levels are expanded -> Button text should say "Expand All" **OK**
> * Verify by clicking on "Expand All" button -> All the levels should be expanded with indentation **OK**
> * Verify button text when all the levels are expanded -> Button text should say "Collapse All" **OK**
> * Verify by clicking on "Collapse All" button -> All but top 2 levels should be collapsed **OK**
> * Verify button text when all 1st level properties are collapsed -> Button text should say "Collapse All" **NO** (Button says "Expand All," as it should in my opinion) See: https://drive.google.com/file/d/1eSE_xpsw86tZpja8hDWFbXlNo7qsFghT/view
> * Verify button text when > *top 2 level properties are collapsed -> Button text should say "Collapse All" **OK**
> * Verify button text when > *top 2 level properties are expanded -> Button text should say "Expand All" **NO** (I'm seeing "Collapse All" at times when 3rd level are expanded) See: https://drive.google.com/file/d/1eSE_xpsw86tZpja8hDWFbXlNo7qsFghT/view
> 
> But, honestly, I'm not sure if those two above that fail make sense to me.

@wooldridge  - The ones that are failing are exactly how they should work as per the JIRA ticket.
If you collapse all 1st level props then it should say 'Expand All' only.

@bsrikan - Looks like we need to update the test plan accordingly. We can have a call to understand it better if needed.

**Comment by bsrikan** [comment]
Checking now.

**Comment by bsrikan** [comment]
@wooldridge and @xnikhil08 I have updated the test plan to sound simpler. 5th scenario("Verify button text when all 1st level properties are collapsed ") should say "Expand All". Changed the same in test plan. You see "Collapse All" sometimes, because of the requirement "In all other cases, you can toggle between Expand All and Collapse All"

For an edge case where we expect "Expand All" to show, we might see "Collapse All" because of the toggle requirement. To me this goes per requirement if not logically.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.tsx [hunk 1]
```diff
@@ -954,20 +1072,23 @@ const SourceToEntityMap = (props) => {
                             </div>
                             :
                             <div id="dataPresent">
-                                <div className={styles.navigationCollapseButtons}>{navigationButtons}</div>
+                                
+                                <div className={styles.navigationCollapseButtons}><span><Button data-testid="expandCollapseBtn-rowKey" onClick={() => handleExpandCollapse('rowKey')} className={styles.expandCollapseBtn}>{expandedSourceFlag ? 'Collapse All' : 'Expand All'}</Button></span><span>{navigationButtons}</span></div>
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.tsx:1076]: Can we update the testid to say expandCollapseBtn-source and expandCollapseBtn-entity so its clear than rowKey and key.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx [hunk 2]
```diff
@@ -390,6 +398,79 @@ describe('RTL Source-to-entity map tests', () => {
...
+        expect(getByText('FirstNamePreferred')).toBeInTheDocument();
+        expect(getByText('LastName')).toBeInTheDocument();
+
+        let expandCollapseBtn = getByTestId('expandCollapseBtn-rowKey');
+
+        expect(expandCollapseBtn.textContent).toBe('Expand All'); // Validating the button label 
+
+        fireEvent.click(expandCollapseBtn); //Expanding all nested levels
+        expect(expandCollapseBtn.textContent).toBe('Collapse All'); // Validating the button label 
+        expect(getByText('suffix')).toBeInTheDocument();
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx:420]: can we add an assertion for indentation check. Something similar to this used in this test file:
expect(getByText('suffix')).toHaveStyle("line-height: 2vh; text-indent: 20px;"); as applicable

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx [hunk 3]
```diff
@@ -390,6 +398,79 @@ describe('RTL Source-to-entity map tests', () => {
...
+
+        fireEvent.click(expandCollapseBtn); //Collapsing back to the default view (root and 1st level)
+        expect(expandCollapseBtn.textContent).toBe('Expand All'); // Validating the button label 
+        expect(getByText('suffix').closest('tr')?.style.display).toBe('none'); // Checking if the row is marked hidden in DOM. All collapsed rows are marked hidden(display: none) once you click on Collapse All button.
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx:424]: can you try using onClosestTableRow from the test-util.ts
expect(onClosesTableRow(getByText('suffix')).style.display).toBe('none')

**Comment by bsrikan** [comment]
Tests look good. We might want to add the scenario in the screencast later after we discuss on that, if its a bug.

**Comment by xnikhil08** [comment]
> Source table seems too close to the toggle button.
> 
> image

@bsrikan - Could you please confirm if you still see this? I can't seem to reproduce this issue at my end.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx (hunk 3 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx]: Done.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx (hunk 2 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx:420]: Done

File: marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.tsx (hunk 1 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.tsx]: Done
